src/ProjectOrigin.Vault/Activities/AllocateActivity.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using MassTransit;
+using Microsoft.Extensions.Logging;
+using ProjectOrigin.Common.V1;
+using ProjectOrigin.PedersenCommitment;
+using ProjectOrigin.Registry.V1;
+using ProjectOrigin.Vault.Database;
+using ProjectOrigin.Vault.Extensions;
+using ProjectOrigin.Vault.Models;
+
+namespace ProjectOrigin.Vault.Activities;
+
+public record AllocateArguments
+{
+    public required Guid AllocationId { get; init; }
+    public required Guid ConsumptionSliceId { get; init; }
+    public required Guid ProductionSliceId { get; init; }
+    public Guid? ChroniclerRequestId { get; init; }
+    public required FederatedStreamId CertificateId { get; init; }
+    public required Guid RequestId { get; init; }
+    public required string Owner { get; init; }
+}
+
+public class AllocateActivity : IExecuteActivity<AllocateArguments>
+{
+    private readonly ILogger<AllocateActivity> _logger;
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IEndpointNameFormatter _formatter;
+
+    public AllocateActivity(
+        ILogger<AllocateActivity> logger,
+        IUnitOfWork unitOfWork,
+        IEndpointNameFormatter formatter)
+    {
+        _logger = logger;
+        _unitOfWork = unitOfWork;
+        _formatter = formatter;
+    }
+
+    public async Task<ExecutionResult> Execute(ExecuteContext<AllocateArguments> context)
+    {
+        try
+        {
+            var cons = await _unitOfWork.CertificateRepository.GetWalletSlice(context.Arguments.ConsumptionSliceId);
+            var prod = await _unitOfWork.CertificateRepository.GetWalletSlice(context.Arguments.ProductionSliceId);
+
+            byte[]? chroniclerSignature = context.Arguments.ChroniclerRequestId is not null
+                ? context.GetVariable<byte[]>(context.Arguments.ChroniclerRequestId.Value.ToString())
+                : null;
+
+            var allocatedEvent = CreateAllocatedEvent(context.Arguments.AllocationId, cons, prod, chroniclerSignature);
```

Comment:
I thought allocation was just something in the wallet but we're sending allocation to the registry now. Why is that?

Reply:
Always have been ;)

Reply:
What are these allocation events? Maybe we should take a call